lib/test/Test/Pos/ConstantsSpec.hs

```diff
@@ -21,15 +20,11 @@ import           Test.Pos.Configuration   (withDefUpdateConfiguration)
 -- @cardano-sl-1.0.4@, something has gone wrong.
 systemTagCheck :: HasUpdateConfiguration => Expectation
 systemTagCheck = do
-    sysTags <- mapM (either error return . mkSystemTag) ["linux64", "macos64", "win64"]
-    let felem = flip elem
+    let sysTags = map SystemTag ["linux64", "macos64", "win64"]
+        felem = flip elem
     ourSystemTag `shouldSatisfy` felem sysTags
 
 spec :: Spec
 spec = withDefUpdateConfiguration $ describe "Constants" $ do
     describe "Configuration constants" $ do
         it "currentSystemTag" $ systemTagCheck
-    describe "UpdateConstants" $ do
-        it "genesisAppNames" $ do
-            for_ C.genesisAppNames $ \(_, name) ->
-                name `shouldSatisfy` isRight
```

Comment:
This test shouldn't be removed, it should check that `checkSystemTag` passes for all `genesisAppNames`.

Reply:
Ok I had a look at this. `C.genesisAppNames` is a constant.

```Haskell
genesisAppNames = map f ["cardano-sl", "csl-daedalus"]
  where
    f name = (name, ApplicationName name)
```

So this test makes no sense.

Reply:
Further, why were we ever testing this? We have these 2 known app names, so we can be fairly (100%) confident that they will be valid `ApplicationName`s.

Reply:
Yes, `C.genesisAppNames` is a constant. Is it a problem?
Testing that `checkSystemTag` passes for genesis names makes sense. This test will fail if someone decreases maximal length of `ApplicationName`, for example.
And this test is very easy to write and executing it should take less than a millisecond, so I don't see a reason to remove it.

Reply:
> This test will fail if someone decreases maximal length of ApplicationName, for example.

Is this something that could be proposed via the update system?

Reply:
No, it can't. So we want this test to fail if it ever happens.

Reply:
This test asserts that genesis application names are valid. This is a very reasonable statement.

Reply:
Hm. Are we ever allowed to change the genesis app names?

Reply:
No.

Reply:
So there's 2 things that aren't allowed to change, and we want to test that they play well together.

Reply:
Wouldn't it be better to have these 2 tests:

- That `checkApplicationName` passes for any string under a given length (quickcheck it).
- That the genesis application names are "cardano-sl" and "csl-daedalus".

Reply:
i.e. to test that the things that cannot change in fact have not changed?

Reply:
Yes, these tests (which you proposed) would be good to have and they are better than this one. But this one is better than nothing. So I don't like that you silently delete some test. I don't mind replacing it with what you proposed.